provider/cluster_waiter_resource.go

```diff
@@ -123,12 +97,43 @@ func (r *ClusterWaiterResource) Create(ctx context.Context,
 
 func (r *ClusterWaiterResource) Read(ctx context.Context, request tfsdk.ReadResourceRequest,
 	response *tfsdk.ReadResourceResponse) {
-	// Do Nothing
+	state := &ClusterWaiterState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	state, err := r.startPolling(ctx, state)
+
+	if err != nil {
+		response.Diagnostics.AddError("Can't poll cluster state", err.Error())
+		return
+	}
+
+	// Save the state:
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
 }
 
 func (r *ClusterWaiterResource) Update(ctx context.Context, request tfsdk.UpdateResourceRequest,
 	response *tfsdk.UpdateResourceResponse) {
-	// Do Nothing
+	plan := &ClusterWaiterState{}
+	diags := request.Plan.Get(ctx, plan)
+	_ = request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	state, err := r.startPolling(ctx, plan)
+
+	if err != nil {
+		response.Diagnostics.AddError("Can't poll cluster state", err.Error())
```

Comment:
Please add here information if this is during create or update
Something like:
```suggestion
		response.Diagnostics.AddError("Can't poll cluster state (Update resource)", err.Error())
```

Reply:
done